tests/test_estimated_degree.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import absolute_import, print_function, division
+import contextlib
+import logging
+
+import pytest
+
+import ufl
+from tsfc import compile_form
+from tsfc.logging import logger
+
+
+class MockHandler(logging.Handler):
+    def emit(self, record):
+        raise RuntimeError()
+
+
+@contextlib.contextmanager
+def push_handler(logger, handler):
+    logger.addHandler(handler)
+    yield
+    logger.removeHandler(handler)
```

Comment:
I don't think you're right. Author of a context manager decides what is right thing to do, here `finally: logger.removeHandler(handler)`, ans its client can just seamlessly use it, not seeing the try-finally clause. There are many different scenarios which would have sense in other cases and it is well documented, see https://docs.python.org/3/library/contextlib.html. `contextlib.contextmanager` is a general tool which requires the dev being explicit without hidden magic. Other wrappers in the package have some implicit error handling.